### DISCUSSION TITLE
markerGenes

### DISCUSSION THREAD

**Discussion by jacobHog** [GreenleafLab/ArchR#1046] [Feature Requests]
Hello,

I was wondering if instead of creating my own list of  "markerGenes" for plotEmbedding(name), is it somehow possible to return the most significant marker genes from getFeatures() and use those as "markerGenes" in plotEmbedding? 
My idea is that the most significant genes would be either above a threshold (e.g. log2(Normcounts+1) = 2) or the top 100 genes with the highest log2(Normcounts+1)?

My reference genome is mm10 with about 28000 genes, and I created a list of marker ~90 genes by scraping the internet. My concern is that I might miss out on important marker genes that I did not include in my own list. 

https://www.archrproject.com/bookdown/marker-genes-imputation-with-magic.html for reference.
https://www.archrproject.com/reference/plotEmbedding.html
https://www.archrproject.com/reference/getFeatures.html

**Comment by rcorces** [comment]
Thanks for posting! In general this type of analysis is performed by manually manipulating the objects that ArchR gives you and we don't provide explicit recommendations or code bits to do it. But you should be able to play around with the return of getMarkerFeatures() to do this with some basic R code.

**Reply by jacobHog** [reply]
Thanks for reply.
I will give it a shot, and post the solution here, if I get that far.

**Reply by jacobHog** [reply]
Hello, I have a follow up question after looking further into the material.

I have generated a geneScoreMatrix by using: https://www.archrproject.com/reference/addGeneScoreMatrix.html as I assume this function computes the gene activity. 
Saved as: AG <- addGeneScoreMatrix(input = ArchrProj, force = TRUE)
Next, I see that AG$ contains some columns, but I am not quite sure which one(s) to compute? I see that ReadsInBlacklist contains measures for biased regions, but where are the values used to generate the gene scoring plots in plotEmbeddings?

**Reply by rcorces** [reply]
From your question, I think theres a bit of a fundamental misunderstanding about how ArchR works. `addGeneScoreMatrix()` returns an `ArchRProject` object.
